### PR TITLE
Add Terratest plumbing

### PR DESCRIPTION
**Pull request by dynamike**
Adds Terratest Plumbing to the module that plumbs a simple test. This PR also adds
* Makefile
* dependabot config
* golangci config
* pre-commit and circleci updates

```
--- PASS: TestTerraformAwsWaf (192.61s)
PASS
ok  	github.com/trussworks/terraform-aws-waf/test	192.784s
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ repos:
       - id: shell-lint
 
   - repo: git://github.com/antonbabenko/pre-commit-terraform
-    rev: v1.19.0
+    rev: v1.21.0
     hooks:
       - id: terraform_docs
       - id: terraform_fmt
```
Comment (kilbergr) [inline .pre-commit-config.yaml]: I believe you need to add
```
- repo: git://github.com/golangci/golangci-lint
    rev: v1.21.0
    hooks:
      - id: golangci-lint
        entry: golangci-lint run --verbose
        verbose: true
```

File: examples/simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#
+# WAF
+#
+
+resource "aws_wafregional_rate_based_rule" "ipratelimit" {
```
Comment (kilbergr) [inline examples/simple/main.tf]:
```suggestion
resource "aws_waf_regional_rate_based_rule" "ipratelimit" {
```

It's a nitpick but I don't understand why there isn't an underscore there

File: examples/simple/main.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+module "waf" {
+  source = "../../"
+
+  alb_arn       = aws_lb.main.arn
+  associate_alb = true
+
+  blocked_path_prefixes = ["/admin", "/password"]
+  allowed_hosts         = ["apples", "oranges"]
+
+  rate_based_rules = [aws_wafregional_rate_based_rule.ipratelimit.id]
```
Comment (kilbergr) [inline examples/simple/main.tf]:
```suggestion
  rate_based_rules = [aws_waf_regional_rate_based_rule.ipratelimit.id]
```
same as above

File: examples/simple/main.tf (hunk 2 above)
Comment (dynamike) [reply examples/simple/main.tf]: That's how the terraform AWS provider names it, I can't change this.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.DS_Store
+.terraform
+terraform.tfstate
```
[the remark this answers is not in the document]
Comment (dynamike) [reply .gitignore]: I need this because it will leave behind 
```
examples/simple/terraform.tfstate
```

**Pull request merged**